kvserver: repro branch for restore2tb snapshots

tbg opened the pull request:
- status quo attempt
- cli: walk directories in debug merge-logs
- cli: default program-filter for merge-logs to match everything
- fatal on MsgApp-induced split

tbg committed:
status quo attempt

tbg committed:
cli: walk directories in debug merge-logs

This is a UX improvement. Instead of typing out

`./cockroach debug merge-logs nodes/*/logs/*.log`

(as one does for each `debug.zip` or roachtest investigation),
you can now simply use

`./cockroach debug merge-logs .`.

Release note: None

tbg committed:
cli: default program-filter for merge-logs to match everything

The program filter is what matches the
`<program>.<host>.<morestuff>.log` in the log files handed to
`merge-logs`. The default is generally a reasonable choice but when you
have an oddly named binary (does happen) merge-logs completely stops
doing anything and time is wasted figuring out why. `merge-logs` is
already very unlikely to pick up stray files since its default
`file-pattern` matches the entire logging file format (it delegates only
matching of `<program>` to the program-filter), so turning the program
filter off by default seems right.

Release note: None

tbg committed:
fatal on MsgApp-induced split

And extra logging for timings.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/63784)
<!-- Reviewable:end -->

tbg committed:
show "age" of split during pre-apply

This should make it easier to search to where things first started
slowing down.

Release note: None

tbg committed:
more logging

Release note: None

tbg committed:
log behind count during split delay

Release note: None

tbg committed:
log verbose on msgapp suppressions

Release note: None

tbg committed:
don't move replicas, only leases, in scatter

Release note: None

tbg committed:
better verbose dropped MsgApp logging

Release note: None

tbg committed:
log when raft snapshot initiated

Release note: None

tbg committed:
skip learner raft snaps earlier

Release note: None

tbg committed:
log ground truth for underreplicated metric

Release note: None

tbg committed:
do not keep clusters

tbg commented:
https://github.com/cockroachdb/cockroach/pull/64060
